tests/includes/juju.sh

```diff
@@ -189,3 +197,12 @@ wait_for() {
         sleep 5
     done
 }
+
+introspect_controller() {
+    local name
+
+    name=${1}
+
+    juju ssh -m controller 0 bash -lic "juju_engine_report" > "${TEST_DIR}/${name}-juju_engine_report.txt"
```

Comment:
This seems a bit flakey given that it is possible that the controller doesn't have a machine 0, or may be in HA mode.

Reply:
What options do we have, quite happy to change... Also if it fails it doesn't cause the destroy to fail.